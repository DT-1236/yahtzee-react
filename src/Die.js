import React, { Component } from 'react';
import styled from 'styled-components';

const Button = styled.button`
  font-size: 25px;
  display: inline-block;
  padding: 0.5em 0.8em;
  font-family: monospace;
  font-weight: bold;
  background-color: red;
  color: white;
  margin-right: 1em;

  &.Die-locked {
    background-color: darkred;
  }
`;

class Die extends Component {
  render() {
    return (
      <Button
        className={this.props.locked ? 'Die Die-locked' : 'Die'}
        onClick={() => this.props.handleClick(this.props.idx)}
      >
        {this.props.val}
      </Button>
    );
  }
}

export default Die;
