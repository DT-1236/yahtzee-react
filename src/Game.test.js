import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Game from './Game';
import Die from './Die';
import { largeStraight } from './Rules';

describe('simple smoke tests', () => {
  it('renders without crashing', () => {
    shallow(<Game />);
  });

  it('mounts without crashing', () => {
    mount(<Game />);
  });
});

describe('simple snapshots', () => {
  it('matches the shallow snapshot', () => {
    const wrapper = shallow(<Game />);
    wrapper.setState({ dice: [1, 1, 1, 1, 1, 1] });
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});

describe('functional testing', () => {
  it('locks die when clicked', () => {
    const wrapper = mount(<Game />);
    expect(wrapper.find('.Die-locked')).toHaveLength(0);
    let die = wrapper.find(Die).at(0);
    die.simulate('click');
    die = wrapper.find(Die).at(0);
    expect(die.props()).toHaveProperty('locked', true);
  });

  it('starts with dice already rolled', () => {
    const wrapper = mount(<Game />);
    let die = wrapper.find(Die).at(0);
    expect(die.props()).not.toHaveProperty('val', undefined);
  });
});

describe('rerolling tests', () => {
  var wrapper;
  var reroll;
  beforeEach(() => {
    // Initialize game with dice
    wrapper = mount(<Game />);
    reroll = wrapper.find('button.Game-reroll');
    // reroll.simulate('click');
  });

  it('should lock all dice after the second reroll', () => {
    expect(wrapper.find('button.Die-locked')).toHaveLength(0);
    reroll.simulate('click');
    reroll.simulate('click');
    expect(wrapper.find('button.Die-locked')).toHaveLength(5);
  });

  it('should disallow rerolls past the second even if dice have been unlocked', () => {
    reroll.simulate('click');
    reroll.simulate('click');
    const locked = wrapper.find('button.Die-locked');
    locked.at(0).simulate('click');
    locked.at(1).simulate('click');
    locked.at(2).simulate('click');
    expect(wrapper.find('button.Die-locked')).toHaveLength(2);
    const previous = wrapper.state().dice;
    reroll.simulate('click');
    expect(wrapper.state().dice).toEqual(previous);
  });
});

describe('score testing', () => {
  it('saves a score when clicked', () => {
    const wrapper = mount(<Game />);
    const ones = wrapper.find('[name="Ones"]');
    expect(wrapper.state().scores).toHaveProperty('ones', undefined);
    ones.simulate('click');
    expect(wrapper.state().scores.ones).not.toEqual(undefined);
  });

  it('does not do anything after a used rule is clicked', () => {
    const wrapper = mount(<Game />);
    const ones = wrapper.find('[name="Ones"]');
    ones.simulate('click');
    const previous = wrapper.state().dice;
    ones.simulate('click');
    // Checks that it does not reroll
    expect(wrapper.state().dice).toEqual(previous);
  });
});
