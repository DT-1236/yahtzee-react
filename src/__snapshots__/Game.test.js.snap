// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple snapshots matches the shallow snapshot 1`] = `
<section>
  <Dice
    dice={
      Array [
        1,
        1,
        1,
        1,
        1,
        1,
      ]
    }
    handleClick={[Function]}
    locked={
      Array [
        false,
        false,
        false,
        false,
        false,
      ]
    }
  />
  <styled.button
    className="Game-reroll"
    disabled={false}
    onClick={[Function]}
  >
    2
     Rerolls Left
  </styled.button>
  <ScoreTable
    doScore={[Function]}
    scores={
      Object {
        "chance": undefined,
        "fives": undefined,
        "fourOfKind": undefined,
        "fours": undefined,
        "fullHouse": undefined,
        "largeStraight": undefined,
        "ones": undefined,
        "sixes": undefined,
        "smallStraight": undefined,
        "threeOfKind": undefined,
        "threes": undefined,
        "twos": undefined,
        "yahtzee": undefined,
      }
    }
  />
</section>
`;
